### PR TITLE
Restore test workarounds for the MSVC-internal Contest infrastructure

### PR DESCRIPTION
**Pull request by StephanTLavavej**
Our MSVC-internal "Contest" infrastructure uses some machines running older OSes, which don't understand ICU and UTF-8 locales. We had removed test workarounds in #2791 (2022-06-15), operating under the belief that this infrastructure would be updated soon. That hasn't happened yet. The affected tests are showing up as "flaky test failures" whenever any MSVC devs run the `std` or `libcxx` suites in Contest.

This PR contains the test workarounds that I've been temporarily applying whenever I mirror GitHub PRs to MSVC. (It's a bit more than just reverting the relevant changes in #2791, as we've had an LLVM update since then.) Note that the changes to `tests/libcxx/skipped_tests.txt` occur at the bottom, which is an MSVC-only section, and therefore are *not* replicated to `expected_results.txt` (which is for the GitHub/Python harness).

Fixes VSO-1601209.

### CODE REVIEW

File: tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp
Comment (barcharcraz) [inline tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp:1123]: is there some internal ticket for fixing this we can reference?

**Review by barcharcraz — APPROVED** [review]

File: tests/std/include/timezone_data.hpp
Comment (StephanTLavavej) [reply tests/std/include/timezone_data.hpp:166]: In principle, I agree. Our ancient `<pmretvals.h>` infrastructure had different exit codes for different reasons, like "skip" (closest in meaning to "unsupported"), and "cascade" versus "fail" (with "cascade" meaning "couldn't even run", usually due to compiler errors). In practice, although I believe the MSVC-internal harness still respects the old codes, I don't see what returning "unsupported"/"skip" would improve for us.

File: tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp
Comment (StephanTLavavej) [reply tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp:1123]: Good idea. I've filed internal [IcM-349538021](https://portal.microsofticm.com/imp/v3/incidents/details/349538021/home) "Outdated OSes on private Contest runners" to track this.

We haven't cited IcM "incidents" in the source code before; given your approval I will assume that the TRANSITION comments, git/PR history, and the filed incident are sufficient for us to remember what happened here.

File: tests/libcxx/skipped_tests.txt
Comment (CaseyCarter) [inline tests/libcxx/skipped_tests.txt:1598]: I'll add `(the Windows version on Contest VMs doesn't always understand ".UTF-8")` here. This is plenty clear enough for folks who've been through this mess twice, but I think inscrutable to new maintainers.

**Review by CaseyCarter — APPROVED** [review]

File: tests/libcxx/skipped_tests.txt
Comment (StephanTLavavej) [reply tests/libcxx/skipped_tests.txt:1598]: Agreed, thank you!

File: tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp
Comment (StephanTLavavej) [reply tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp:1123]: Do you want it in the MSVC-PR only, or also the GH PR?

Comment (CaseyCarter) [reply tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp:1123]: I'd prefer both.

Comment (CaseyCarter) [reply tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp:1123]: Welp, now that the IcM ticket has been closed and Contest supposedly decommissioned, shall we abandon this PR?

Comment (StephanTLavavej) [reply tests/std/tests/P0355R7_calendars_and_time_zones_formatting/test.cpp:1123]: It's not totally decommissioned, as apparently some LKG test runs are still running Contest-Private checks, but I'd be happy to abandon this PR. We've finally officially Gotten The Memo about running `runall-private.ps1`, I've verified that it succeeds without workarounds for x86 and x64, and we've clearly survived for months with the status quo. (Reportedly, Phase 6 will truly put an end to Contest.)

**Pull request closed without merging**
